### ISSUE TITLE
💅 `useExhaustiveDependencies` incorrectly detects non-destructured stable functions

### ISSUE THREAD

**Issue by ShadowWolf308** [biomejs/biome#8484]
### Environment information

```bash
CLI:
  Version:                      2.3.9
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.20.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.18.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
  HTML full support enabled:    unset

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useExhaustiveDependencies
    correctness/useIsNan
    correctness/useParseIntRadix
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/noProto
    nursery/noScriptUrl
    nursery/useSpread
    performance/noAccumulatingSpread
    performance/noDynamicNamespaceImportAccess
    security/noBlankTarget
    security/noGlobalEval
    style/noDescendingSpecificity
    style/noNonNullAssertion
    style/useArrayLiterals
    style/useConsistentArrayType
    style/useConsistentMemberAccessibility
    style/useConst
    style/useDeprecatedReason
    style/useExponentiationOperator
    style/useExportType
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodejsImportProtocol
    style/useShorthandFunctionType
    style/useTemplate
    suspicious/noApproximativeNumericConstant
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExplicitAny
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/useExhaustiveDependencies

### Playground link

https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAdQBzAGUATQBlAG0AbwAgAH0AIABmAHIAbwBtACAAIgByAGUAYQBjAHQAIgA7AAoACgAvAC8AIABgAHUAcwBlAFMAdABhAGIAbABlAGAAIABpAHMAIABtAGEAcgBrAGUAZAAgAGkAbgAgAHQAaABlACAAYABvAHAAdABpAG8AbgBzAC4AaABvAG8AawBzAGAAIABmAG8AcgAgAGAAdQBzAGUARQB4AGgAYQB1AHMAdABpAHYAZQBEAGUAcABlAG4AZABlAG4AYwBpAGUAcwBgACAAYQBzADoACgAvAC8AIAB7AAoALwAvACAAIAAgACAAIAAiAG4AYQBtAGUAIgA6ACAAIgB1AHMAZQBTAHQAYQBiAGwAZQAiACwACgAvAC8AIAAgACAAIAAgACIAcwB0AGEAYgBsAGUAUgBlAHMAdQBsAHQAIgA6ACAAWwAiAHMAdABhAGIAbABlAFYAYQBsAHUAZQAiAF0ACgAvAC8AIAB9AAoACgBmAHUAbgBjAHQAaQBvAG4AIAB1AHMAZQBTAHQAYQBiAGwAZQAoACkAIAB7AAoAIAAgAGMAbwBuAHMAdAAgAHMAdABhAGIAbABlACAAPQAgAHUAcwBlAFIAZQBmACgAIgBTAG8AbQBlACAAcwB0AGEAYgBsAGUAIAB2AGEAbAB1AGUAIgApADsACgAgACAACgAgACAAcgBlAHQAdQByAG4AIAB7AAoAIAAgACAAIAB2AGEAbAB1AGUAOgAgACIAUwBvAG0AZQAgAHYAYQBsAHUAZQAiACwACgAgACAAIAAgAHMAdABhAGIAbABlAFYAYQBsAHUAZQA6ACAAcwB0AGEAYgBsAGUALgBjAHUAcgByAGUAbgB0ACwACgAgACAAfQAKAH0ACgAKAGYAdQBuAGMAdABpAG8AbgAgAF8AQwBvAG0AcAAoACkAIAB7AAoAIAAgAGMAbwBuAHMAdAAgAHMAdABhAGIAbABlAE8AYgBqACAAPQAgAHUAcwBlAFMAdABhAGIAbABlACgAKQA7AAoACgAgACAALwAvACAAZwBpAHYAZQBzACAAZQByAHIAbwByACAAYgB1AHQAIABlAHgAcABlAGMAdABzACAAbgBvACAAZQByAHIAbwByAAoAIAAgAC8ALwAgAFQAaABlACAAdgBhAGwAdQBlACAAdABoAGUAIABlAHIAcgBvAHIAIABpAHMAIABhAGIAbwB1AHQAIABpAHMAIABzAHQAYQBiAGwAZQAKACAAIABjAG8AbgBzAHQAIABfAHYAYQBsAHUAZQAgAD0AIAB1AHMAZQBNAGUAbQBvACgAKAApACAAPQA%2BACAAewAKACAAIAAgACAAaQBmACAAKABzAHQAYQBiAGwAZQBPAGIAagAuAHYAYQBsAHUAZQAgAD0APQA9ACAAcwB0AGEAYgBsAGUATwBiAGoALgBzAHQAYQBiAGwAZQBWAGEAbAB1AGUAKQAgAHsACgAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgAHMAdABhAGIAbABlAE8AYgBqAC4AcwB0AGEAYgBsAGUAVgBhAGwAdQBlADsAIAAgAAoAIAAgACAAIAB9AAoACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAcwB0AGEAYgBsAGUATwBiAGoALgB2AGEAbAB1AGUAOwAKACAAIAB9ACwAIABbAHMAdABhAGIAbABlAE8AYgBqAC4AdgBhAGwAdQBlAF0AKQA7AAoAIAAgAAoAIAAgAHIAZQB0AHUAcgBuACAAbgB1AGwAbAA7AAoAfQAKAAoAZgB1AG4AYwB0AGkAbwBuACAAXwBDAG8AbQBwADIAKAApACAAewAKACAAIABjAG8AbgBzAHQAIAB7ACAAdgBhAGwAdQBlACwAIABzAHQAYQBiAGwAZQBWAGEAbAB1AGUAIAB9ACAAPQAgAHUAcwBlAFMAdABhAGIAbABlACgAKQA7AAoACgAgACAALwAvACAAVwBvAHIAawBzACAAYQBzACAAZQB4AHAAZQBjAHQAZQBkAAoAIAAgAGMAbwBuAHMAdAAgAF8AdgBhAGwAdQBlACAAPQAgAHUAcwBlAE0AZQBtAG8AKAAoACkAIAA9AD4AIAB7AAoAIAAgACAAIABpAGYAIAAoAHYAYQBsAHUAZQAgAD0APQA9ACAAcwB0AGEAYgBsAGUAVgBhAGwAdQBlACkAIAB7AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIABzAHQAYQBiAGwAZQBWAGEAbAB1AGUAOwAgACAACgAgACAAIAAgAH0ACgAgACAAIAAgAAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAHYAYQBsAHUAZQA7AAoAIAAgAH0ALAAgAFsAdgBhAGwAdQBlAF0AKQA7AAoAIAAgAAoAIAAgAHIAZQB0AHUAcgBuACAAbgB1AGwAbAA7AAoAfQA%3D&ruleDomains.react=recommended

### Expected result

It gives an error when the stable value is not destructured but if it is not destructured it is still a stable value and doesn't have to be added to the deps array, when it is destructured it works as expected

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
